File: src/main/java/factorio/debugger/DAP/messages/DAPAdditionalProperties.java
package factorio.debugger.DAP.messages;

import java.util.HashMap;
import java.util.Map;
import com.fasterxml.jackson.annotation.JsonAnyGetter;
import com.fasterxml.jackson.annotation.JsonAnySetter;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonInclude;

@JsonInclude(JsonInclude.Include.NON_EMPTY)
public class DAPAdditionalProperties {
    @JsonIgnore
    public Map<String, Object> additionalProperties = new HashMap<>();

    // Capture all other fields that Jackson do not match other members
    @JsonAnyGetter
    public Map<String, Object> otherFields() {
        return additionalProperties;
    }

    @JsonAnySetter
    public void setOtherField(String name, Object value) {
        additionalProperties.put(name, value);
    }
}

File: src/main/kotlin/factorio/debugger/DAP/messages/DAPAdditionalProperties.kt
package factorio.debugger.DAP.messages

import com.fasterxml.jackson.annotation.JsonAnyGetter
import com.fasterxml.jackson.annotation.JsonAnySetter
import com.fasterxml.jackson.annotation.JsonIgnore
import com.fasterxml.jackson.annotation.JsonInclude

@JsonInclude(JsonInclude.Include.NON_EMPTY)
open class DAPAdditionalProperties {
    @JvmField
    @JsonIgnore
    var additionalProperties: MutableMap<String, Any> = HashMap()

    // Capture all other fields that Jackson do not match other members
    @JsonAnyGetter
    fun otherFields(): Map<String, Any> {
        return additionalProperties
    }

    @JsonAnySetter
    fun setOtherField(name: String, value: Any) {
        additionalProperties[name] = value
    }
}
